### PR TITLE
Remove buildkite configs and references

### PR DESCRIPTION
**Pull request by rodrigozhou**
## What changed?

Remove buildkite configs and references.

## Why?

Buildkite is no longer used.

## How did you test it?

Existing tests.

## Potential risks

No.

## Documentation

N/A

## Is hotfix candidate?

No.

### CODE REVIEW

File: common/persistence/tests/postgresql_test.go [hunk 1]
```diff
@@ -176,23 +176,12 @@ func (p *PostgreSQLSuite) TestPostgreSQLClusterMetadataPersistence() {
...
-// 	s.TestBase = persistencetests.NewTestBaseWithSQL(persistencetests.GetPostgreSQLTestClusterOption())
-// 	s.TestBase.Setup(nil)
-// 	suite.Run(p.T(), s)
-// }
+func (p *PostgreSQLSuite) TestPostgreSQLQueuePersistence() {
+	s := new(persistencetests.QueuePersistenceSuite)
+	s.TestBase = persistencetests.NewTestBaseWithSQL(persistencetests.GetPostgreSQLTestClusterOption())
+	s.TestBase.Setup(nil)
+	suite.Run(p.T(), s)
+}
```
Comment (rodrigozhou) [inline common/persistence/tests/postgresql_test.go:184]: Trying to uncomment this test marked as flaky in buildkite. Let's see if will be flaky in GHA.

File: Makefile [hunk 2]
```diff
@@ -8,7 +8,7 @@ bins: temporal-server temporal-cassandra-tool temporal-sql-tool tdbg
 # Install all tools, recompile proto files, run all possible checks and tests (long but comprehensive).
 all: update-tools clean proto bins check test
 
-# Used by Buildkite.
+# Used in CI  tests
```
Comment (stephanos) [inline Makefile:11]:
```suggestion
# Used in CI 
```

File: Makefile [hunk 3]
```diff
@@ -383,11 +380,6 @@ coverage-report: $(SUMMARY_COVER_PROFILE)
 	@printf $(COLOR) "Generate HTML report from $(SUMMARY_COVER_PROFILE) to $(SUMMARY_COVER_PROFILE).html..."
 	@go tool cover -html=$(SUMMARY_COVER_PROFILE) -o $(SUMMARY_COVER_PROFILE).html
 
-ci-coverage-report: $(SUMMARY_COVER_PROFILE) coverage-report
-	@printf $(COLOR) "Generate Coveralls report from $(SUMMARY_COVER_PROFILE)..."
-	go install github.com/mattn/goveralls@v0.0.7
-	@goveralls -coverprofile=$(SUMMARY_COVER_PROFILE) -service=buildkite || true
-
```
Comment (stephanos) [inline Makefile:390]: Is the GHA doing this now?

**Review by stephanos — APPROVED** [review]

File: Makefile (hunk 3 above)
Comment (rodrigozhou) [reply Makefile:390]: Not at this moment. Maybe another PR to add it. Also, noticed that the last update to coveralls was in October, which is a bit weird...

**Pull request merged**